src/libraries/System.Runtime.InteropServices.JavaScript/src/System/Runtime/InteropServices/JavaScript/JSFunctionBinding.cs

```diff
@@ -142,8 +142,9 @@ public static void InvokeJS(JSFunctionBinding signature, Span<JSMarshalerArgumen
         /// <exception cref="PlatformNotSupportedException">The method is executed on an architecture other than WebAssembly.</exception>
         // JavaScriptExports need to be protected from trimming because they are used from C/JS code which IL linker can't see
         [DynamicDependency(DynamicallyAccessedMemberTypes.PublicMethods, "System.Runtime.InteropServices.JavaScript.JavaScriptExports", "System.Runtime.InteropServices.JavaScript")]
-        // TODO make this DynamicDependency conditional
+#if FEATURE_LEGACY_JS_INTEROP
         [DynamicDependency(DynamicallyAccessedMemberTypes.PublicMethods, "System.Runtime.InteropServices.JavaScript.LegacyExports", "System.Runtime.InteropServices.JavaScript")]
+#endif
```

Comment:
This does not let us to remove LegacyExports because it's a library compile time setting. We could remove this `DynamicDependency` and use XML descriptor with `System.Runtime.InteropServices.JavaScript.LegacyExports` and pass it to linker only when legacy interop is enabled.

Reply:
That XML would become embedded resource, right ? How do I tell the linker to ignore it or not in `WasmApp.Native.targets` ?

Reply:
It would be additional file passed conditionally to the linker (we run it all the time to this should work).

Reply:
That won't work `DynamicDependency` is always processed and it only removed later. The attribute cannot be visible to linker in this case to work as expected.